pkg/execution/state/redis_state/queue.go

```diff
@@ -1586,7 +1613,7 @@ func (q *queue) Lease(ctx context.Context, p QueuePartition, item QueueItem, dur
 	}
 
 	// Grab all partitions for the queue item
-	parts, acctLimit := q.ItemPartitions(ctx, item)
+	parts, acctLimit := q.ItemPartitions(ctx, item, true)
```

Comment:
We do want to consider key queues on Lease, to add items to all concurrency queues. This is important because we wouldn't apply custom concurrency key limits otherwise.